### PR TITLE
DOC: fix build failure with sphinx 7 and SOURCE_DATE_EPOCH

### PR DESCRIPTION
**Pull request by rebecca-palmer**
When SOURCE_DATE_EPOCH is set (which is the default when building Debian packages), Sphinx 7 [**requires** a space or comma after the year in 'copyright'](https://sources.debian.org/src/sphinx/7.1.2-2/sphinx/config.py/#L444).

[Debian bug report, with failure log](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1042672).

- [not known to be reported here ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ n/a] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ CI says yes] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [n/a ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [no ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -162,7 +162,7 @@
 # General information about the project.
 project = "pandas"
 # We have our custom "pandas_footer.html" template, using copyright for the current year
-copyright = f"{datetime.now().year}"
+copyright = f"{datetime.now().year} "
```
Comment (mroeschke) [inline doc/source/conf.py:165]: Can we use the comma ending instead of the white space?

File: doc/_templates/pandas_footer.html [hunk 2]
```diff
@@ -1,3 +1,3 @@
 <p class="copyright">
-    &copy {{copyright}} pandas via <a href="https://numfocus.org">NumFOCUS, Inc.</a> Hosted by <a href="https://www.ovhcloud.com">OVHcloud</a>.
```
Comment (mroeschke) [inline doc/_templates/pandas_footer.html:2]: Is this still needed if we use the comma ending?

File: doc/source/conf.py (hunk 1 above)
Comment (rebecca-palmer) [reply doc/source/conf.py:165]: You could but that comma would presumably appear in the rendered documentation - my reason for using a space here and removing it from the other file was to not change the rendered documentation

Comment (mroeschke) [reply doc/source/conf.py:165]: I think changing the rendered docs is okay if it simplifies the changes

File: doc/_templates/pandas_footer.html (hunk 2 above)
Comment (rebecca-palmer) [reply doc/_templates/pandas_footer.html:2]: It's not _strictly_ needed _either_ way, but not doing it means you get 2 spaces or comma-then-space where there was previously one space

File: doc/source/conf.py (hunk 1 above)
Comment (rebecca-palmer) [reply doc/source/conf.py:165]: That simplification is literally "don't remove one space", and given that the code is already written, changing it now is more work not less.  But this isn't worth arguing over, so whatever.

File: doc/_templates/pandas_footer.html (hunk 2 above)
Comment (mroeschke) [reply doc/_templates/pandas_footer.html:2]: The comma-then-space format seems OK to me

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rebecca-palmer

**Pull request merged**